pallets/nfts/src/lib.rs

```diff
@@ -418,6 +427,30 @@ pub mod pallet {
 		OptionQuery,
 	>;
 
+	/// Number of collection items owned by an account.
+	#[pallet::storage]
+	pub type AccountBalance<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
```

Comment:
There is currently no deposit protecting this state. Assuming each collection has a single owner for every item, then thats potentially u32::MAX balances per collection which is not accounted for. We could increase the `NftsItemDeposit` value in the runtime to include this storage requirement size, but that is charged per item and not per owner.

Reply:
One possible I can think of is we charge the `collection_owner` `4 bytes * deposit_per_byte` on every record of `AccountBalance(collection, owner)` created (`mint` and `transfer`). With `mint` this is possible but with `transfer`, I am not sure because if there is one entity batch transfer items to multiple accounts, the collection owner have to pay all for them so it would be `4 bytes * deposit_per_byte * (number of item holders)`

Reply:
We take a deposit so an owner is incentivized to clean up state when no longer required. If the item deposit is sufficiently high, then one could argue that it serves the purpose of removing the last owned item, and therefore the last account balance value.

My only concern with fudging it via the runtime deposit value is that the pallet implementation remains flawed. It is up to the runtime dev to ensure this is included and it is not immediately obvious. If we do track it separately via its own deposit, it then becomes a case of tracking the deposit value taken and possibly storing it alongside the balance. There is obvious additional complexity required.

Reply:
#413